modules/streams/arrow-streams/src/main/kotlin/arrow/streams/internal/FreeC.kt

```diff
@@ -0,0 +1,345 @@
+package arrow.streams.internal
+
+import arrow.Kind
+import arrow.core.*
+import arrow.effects.ExitCase
+import arrow.streams.CompositeFailure
+import arrow.streams.internal.FreeC.Result
+import arrow.typeclasses.MonadError
+
+/**
+ * Free Monad with Catch (and Interruption).
```

Comment:
We should mention and include fs2 here https://github.com/arrow-kt/arrow/blob/master/LICENSE#L20